Objective-C/CBLReplicator.mm

```diff
@@ -180,8 +180,12 @@ - (void) _start {
         .callbackContext = (__bridge void*)self,
         // TODO: Add .validationFunc (public API TBD)
     };
-    C4Error err;
-    _repl = c4repl_new(_config.database.c4db, addr, dbName, otherDB.c4db, params, &err);
+    
```

Comment:
This PR isn't supposed to make CBLReplicator thread-safe; that's coming later, right? Because there are still race conditions involving `_repl`.

Reply:
Yes. I haven't made the replicator thread-safe yet.